File: tobehe/100.same-tree.cpp
/*
 * @lc app=leetcode id=100 lang=cpp
 *
 * [100] Same Tree
 */
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if(p==NULL&&q==NULL)
            return true;
        if(p!=NULL&&q!=NULL)
        {
            if(p->val!=q->val)
            {
                return false;
            }
            else
            {
                bool flag1 =  isSameTree(p->left,q->left);
                if(flag1== false)
                {
                    return false;
                }
                else
                {
                    return isSameTree(p->right,q->right);
                }
                
            }
            
        }
        else
        {
            return false;
        }
        

    }
};

File: AnYang/Tree/100. Same Tree.py
# 2019-09-04 Wed

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        if p == None or q == None:
            return True if p == None and q == None else False
        return False if p.val != q.val else self.isSameTree(q.left,p.left) and self.isSameTree(p.right, q.right)